src/bootstrap/src/core/build_steps/setup.rs

```diff
@@ -170,22 +170,13 @@ impl Step for Profile {
     }
 
     fn run(self, builder: &Builder<'_>) {
-        // During ./x.py setup once you select the codegen profile.
-        // The submodule will be downloaded. It does not work in the
-        // tarball case since they don't include Git and submodules
-        // are already included.
-        if !builder.rust_info().is_from_tarball() {
```

Comment:
I don't think there's much benefit from this. Not everyone doing codegen will be modifying LLVM and anyone that wants to will figure out that they need to init the submodule somehow